src/dashboard/store/modules/db.js

```diff
@@ -27,13 +29,45 @@ export default {
   },
 
   getters: {
+    favoriteCreators(state, getters) {
+      let creators = _.orderBy(
+        getters.creatorsNotIgnored,
+        ['priority', 'score'],
+        ['asc', 'desc']
+      );
+      return creators.slice(0, 12);
+    },
     creatorsNotIgnored(state, getters) {
       let creators = _.filter(getters.creators, c => c.ignore !== true);
-      creators = _.orderBy(creators, ['priority', 'duration'], ['asc', 'desc']);
       return creators;
     },
     creators(state) {
-      return _.map(state.creators, (e, i) => ({ ...e, index: i }));
+      return _.map(state.creators, (e, i) => ({
+        ...e,
+        index: i,
+        score: scoringFunction(e),
+      }));
+    },
+    creatorsWithShare(state, getters) {
+      if (getters.favoriteCreators.length > 0) {
+        let creators = getters.favoriteCreators;
+        let totScore = _.sumBy(creators, 'score');
+        let factor = 1 - _.sumBy(creators, 'share');
+
+        return creators.map(c => {
+          let share;
+          if (c.share > 0) {
+            share = c.share;
+          } else {
+            share = (c.score / totScore) * factor;
+          }
+          return {
+            ..._.pick(c, ['name', 'duration', 'url', 'address', 'index']),
```

Comment:
Why not just `...c`?

Reply:
We don't need the other things in `c` we might take the whole thing but I like telling the reader of the code what we're interested in. In any case it's cleaner than what was there before where we basically did pick manually.